strawberry/sanic/views.py

```diff
@@ -0,0 +1,96 @@
+import json
+
+from sanic.exceptions import abort
+from sanic.response import HTTPResponse, html
+from sanic.views import HTTPMethodView
+from strawberry.http import GraphQLHTTPResponse, process_result
+from strawberry.types import ExecutionResult
+
+from ..schema import BaseSchema
+from .graphiql import render_graphiql_page
+
+
+class GraphQLView(HTTPMethodView):
+    """
+    Class based view to handle GraphQL HTTP Requests
+
+    Args:
+        schema: strawberry.Schema
+        graphiql: bool, default is True
+
+    Returns:
+        None
+
+    Example:
+        app.add_route(
+            GraphQLView.as_view(schema=schema, graphiql=True),
+            "/graphql"
+        )
+    """
+
+    methods = ["GET", "POST"]
+
+    def __init__(self, schema: BaseSchema, graphiql: bool = True):
+        self.graphiql = graphiql
+        self.schema = schema
+
+    def get_root_value(self):
+        return None
+
+    def get_context(self, request):
+        return {"request": request}
+
+    def render_template(self, template=None):
+        return html(template)
+
+    def process_result(self, result: ExecutionResult) -> GraphQLHTTPResponse:
+        return process_result(result)
+
+    async def dispatch_request(self, request):
+        request_method = request.method.lower()
+        if not self.graphiql:
+            abort(404)
+
+        show_graphiql = request_method == "get" and self.should_display_graphiql(
+            request
+        )
+
+        if show_graphiql:
+            template = render_graphiql_page()
+            return self.render_template(template=template)
+
+        data = request.json
+
+        try:
+            query = data["query"]
+            variables = data.get("variables")
+            operation_name = data.get("operationName")
+
+        except KeyError:
+            return HTTPResponse("No valid query was provided for the request", 400)
```

Comment:
Can we add a test case on the `KeyError` here for coverage? It also seems as if the try/except should be around just the dict `[]` access and not the `.get` calls

Reply:
Agreed, the changes have been implemented in commit d1e0ced